src/liballoc/benches/lib.rs

```diff
@@ -1,5 +1,6 @@
 #![feature(repr_simd)]
 #![feature(test)]
+#![feature(benches_btree_set)]
```

Comment:
Before merging I think we should remove this feature and make the methods private (or just inline them). It's nice to get an idea of the performance characteristics of each strategy, but once we understand those I think we can move forward with just the general benchmarks.

Reply:
But we only understand the performance characteristics:

- on 1 platform and 1 machine (I could build it on Linux on another machine though, but nothing modern),
- for one generation of the BTreeSet implementation. I see significant improvement from stable to nightly in my BTreeSet macro-benchmarks, tens of percents, with the same intersection implementation.

What alternatives are there for the feature litter?
- Moving the benchmarks closer to the lib code is terrible: every tweak or comment requires over an hour to build.
- Duplicating the lib code near/in the benchmark file is doable. You should notice if it's not up to date anymore when you check that the actual performance matches that of either implementation.
- Or similarly, duplicating the lib code and benchmarks in some separate repository (like I already did in https://github.com/ssomers/Bron-Kerbosch/blob/master/rust/bron_kerbosch/src/util.rs).

> (or just inline them)

I don't understand. Are you saying that Rust has ways to get around the privateness of the enum, without exposing _something_ as public?

Reply:
@ssomers I think duplicating the code in a separate repository is the best way to go here :+1: I _believe_ we can run benchmarks in CI here (I don't remember the command off the top of my head) so we can worry about this later.